Assets/Mirror/Components/NetworkTransformBase.cs

```diff
@@ -16,131 +16,265 @@
+        [SyncVar]
+        public bool interpolatePosition = true;
+
+        [Tooltip("Set to true if rotation should be interpolated")]
+        [SyncVar]
+        public bool interpolateRotation = true;
+
+        [Tooltip("Set to true if scale should be interpolated")]
+        [SyncVar]
+        public bool interpolateScale = true;
 
         // Sensitivity is added for VR where human players tend to have micro movements so this can quiet down
         // the network traffic.  Additionally, rigidbody drift should send less traffic, e.g very slow sliding / rolling.
         [Header("Sensitivity")]
+
         [Tooltip("Changes to the transform must exceed these values to be transmitted on the network.")]
+        [SyncVar]
         public float localPositionSensitivity = .01f;
+
         [Tooltip("If rotation exceeds this angle, it will be transmitted on the network")]
+        [SyncVar]
         public float localRotationSensitivity = .01f;
+
         [Tooltip("Changes to the transform must exceed these values to be transmitted on the network.")]
+        [SyncVar]
         public float localScaleSensitivity = .01f;
 
-        // target transform to sync. can be on a child.
-        protected abstract Transform targetComponent { get; }
+        [Header("Diagnostics")]
 
         // server
-        Vector3 lastPosition;
-        Quaternion lastRotation;
-        Vector3 lastScale;
+        public Vector3 lastPosition;
+        public Quaternion lastRotation;
+        public Vector3 lastScale;
 
         // client
-        public class DataPoint
+        // use local position/rotation for VR support
+        [Serializable]
+        public struct DataPoint
         {
             public float timeStamp;
-            // use local position/rotation for VR support
             public Vector3 localPosition;
             public Quaternion localRotation;
             public Vector3 localScale;
             public float movementSpeed;
+
+            public bool isValid => timeStamp != 0;
         }
+
+        // Is this a client with authority over this transform?
+        // This component could be on the player object or any object that has been assigned authority to this client.
+        bool IsOwnerWithClientAuthority => hasAuthority && clientAuthority;
+
         // interpolation start and goal
-        DataPoint start;
-        DataPoint goal;
+        public DataPoint start = new DataPoint();
+        public DataPoint goal = new DataPoint();
 
-        // local authority send time
-        float lastClientSendTime;
+        // We need to store this locally on the server so clients can't request Authority when ever they like
+        bool clientAuthorityBeforeTeleport;
 
-        // serialization is needed by OnSerialize and by manual sending from authority
-        // public only for tests
-        public static void SerializeIntoWriter(NetworkWriter writer, Vector3 position, Quaternion rotation, Vector3 scale)
+        void FixedUpdate()
```

Comment:
As mentioned in discord we shouldn't use FixUpdated for networking 

We don't want tp skip a network update every 100ms, or send 2 updates 0ms apart
![Time between updates](https://cdn.discordapp.com/attachments/468084409858523146/753734976079659028/unknown.png)